projects/packages/forms/src/blocks/contact-form/editor.scss

```diff
@@ -1065,6 +1066,10 @@
 	font-weight: 700;
 }
 
+:where(.is-style-outlined .wp-block-jetpack-label) {
```

Comment:
This is making the specificity 0-0-0. Elsewhere we're using 0-1-0 specificity for these kind of styles, the rationale being that we want to override css resets which are often 0-0-1.

Hard to say if a theme would change the label font size, but I think a small bump in specificity is safest.